eventhubs/client/azure-eventhubs/src/main/java/com/azure/eventhubs/Retry.java

```diff
@@ -0,0 +1,118 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.eventhubs;
+
+import com.azure.core.amqp.exception.AmqpException;
+import com.azure.core.amqp.exception.ErrorCondition;
+import com.azure.eventhubs.implementation.ClientConstants;
+
+import java.time.Duration;
+import java.util.concurrent.atomic.AtomicInteger;
+
+public abstract class Retry {
+
+    public static final Retry NO_RETRY = new RetryExponential(Duration.ofSeconds(0), Duration.ofSeconds(0), 0);
+
+    private AtomicInteger retryCount = new AtomicInteger(0);
+
+    /**
+     * Check if the existing exception is a retryable exception.
+     *
+     * @param exception A exception that was observed for the operation to be retried.
+     * @return true if the exception is a retryable exception, otherwise false.
+     * @throws IllegalArgumentException when the exception is null.
+     */
+    public static boolean isRetryableException(Exception exception) {
+        if (exception == null) {
+            throw new IllegalArgumentException("exception cannot be null");
+        }
+
+        if (exception instanceof AmqpException) {
```

Comment:
You can simplify this entire block by going:
`return (exception instanceof AmqpException) && ((AmqpException) exception).isTransient()`